documentation/getstarted/advanced/migrationtesting.md

```diff
@@ -0,0 +1,241 @@
+
+    ```
+    #!/bin/bash
+
+    set -e
+
+    echo "Downloading and installing spawnctl..."
+    curl -sL https://run.spawn.cc/install | sh
+    export PATH=$HOME/.spawnctl/bin:$PATH
+
+    export SPAWN_PAGILA_IMAGE_NAME=Pagila:prod
+
+    echo "Creating Pagila backup Spawn data container from image '$SPAWN_PAGILA_IMAGE_NAME'..."
+    pagilaContainerName=$(spawnctl create data-container --image $SPAWN_PAGILA_IMAGE_NAME --lifetime 10m --accessToken $SPAWNCTL_ACCESS_TOKEN -q)
+
+    pagilaJson=$(spawnctl get data-container $pagilaContainerName -o json)
+    pagilaHost=$(echo $pagilaJson | jq -r '.host')
+    pagilaPort=$(echo $pagilaJson | jq -r '.port')
+    pagilaUser=$(echo $pagilaJson | jq -r '.user')
+    pagilaPassword=$(echo $pagilaJson | jq -r '.password')
+
+    echo "Successfully created Spawn data container '$pagilaContainerName'"
+
+    docker run --net=host --rm -v $PWD/sql:/flyway/sql flyway/flyway migrate -url="jdbc:postgresql://$pagilaHost:$pagilaPort/pagila" -user=$pagilaUser -password=$pagilaPassword
+
+    echo "Successfully migrated 'Pagila' database"
+
+    spawnctl delete data-container $pagilaContainerName --accessToken $SPAWNCTL_ACCESS_TOKEN -q
+
+    echo "Successfully cleaned up the Spawn data container '$pagilaContainerName'"
+    ```    
+
+1. Before committing and pushing this file to the repository, lets give it executable permissions:
+
+    <pre class="console"><span>&gt;</span> chmod +x migration-test-prod.sh</pre>
+
+This script accomplishes a few things. We are:
+  * Installing Spawn to give us access to the `spawnctl` command from the pipeline agent
+  * Creating a [Spawn data container](https://www.spawn.cc/docs/concepts-data-container) from the data image we created locally named `Pagila:prod`
+  * Using the official [Flyway docker image](https://hub.docker.com/r/flyway/flyway) to run `flyway migrate` on our data container, using migration scripts stored under the `sql` folder
+  * Automating the cleanup of the database
+
+That it - that is our migration test. We have quickly provisioned a database instance from our back up using Spawn, and set the flyway connection details to point to that database and run the migration scripts in our repository. Any errors will be apparent here and show up on the developers Pull Request that this ran against.
+
+Note: There is a [Flyway Migration action](https://github.com/marketplace/actions/flyway-migration) in the GitHub Marketplace which you can copy from. But using the code in `migrate-test.sh` is using generic bash and will work across all CI pipelines.
+
+### Run Flyway migration tests
+
+Once you've pushed all your changes to GitHub, you can now manually run the migration test workflow by navigating to the Actions tab in GitHub and clicking on 'Database migration test', then 'Run workflow'. Or, a more likely scenario, we want this to automatically run on a PR request so the development team can test that any new migrations will run against master before merging.
```

Comment:
Where it says `PR request` should probably just be `PR`

Reply:
oops, "pull request request" 😳

Reply:
7fed50d